Upgrade to govuk-frontend v5.5.0

AshGDS opened the pull request:
## What/  Why
<!-- What are the reasons behind this change being made? -->
- Upgrade to `govuk-frontend` v5.5.0
- See their CHANGELOG for changes: https://github.com/alphagov/govuk-frontend/blob/main/CHANGELOG.md
- Incorporates this PR's commit in order to fix the search styles: https://github.com/alphagov/govuk_publishing_components/pull/3991 
- This `finder-frontend` PR also relies on this upgrade https://github.com/alphagov/finder-frontend/pull/3334
- I have tested this PR in `finder-frontend` with the two PRs above included, and the search on mobile looked correct

Trello card: https://trello.com/c/l2yxZZ9k/274-upgrade-to-govuk-frontend-54

## Visual Changes
<!-- If change results in visual changes, include detailed screenshots that show the various states. -->

<!-- Please ensure that the changes are reviewed by a Designer if required. -->
<!-- To help Designers, please include a link to specific elements to review, -->
<!-- for example to https://components-gem-pr-[PULL REQUEST NUMBER].herokuapp.com/public -->

### The upgrade changes the typography Scale

<img width="941" alt="image" src="https://github.com/user-attachments/assets/ce1d00a1-db37-4fef-ad37-83df43d67192">
<img width="941" alt="image" src="https://github.com/user-attachments/assets/ef7960b8-170b-47ae-a1cf-dc06caf0e4ef">


### The upgrade adds new organisation colours

<img width="570" alt="image" src="https://github.com/user-attachments/assets/dc63742a-4dbd-4783-8a42-26684694cf0a">
<img width="570" alt="image" src="https://github.com/user-attachments/assets/7a5e458d-4583-4e7a-acb1-27fb8e0f517d">
<img width="560" alt="image" src="https://github.com/user-attachments/assets/8e65a328-b21e-493c-a6aa-251fb8c6a9e1">
<img width="560" alt="image" src="https://github.com/user-attachments/assets/9b31060c-2b5e-4018-878d-bde78bf79594">

### The upgrade fixes the checkbox component's conditional reveal email address alignment

<img width="489" alt="image" src="https://github.com/user-attachments/assets/d3cfb3d3-ad02-4ea6-8318-31c11ed204b1">
<img width="489" alt="image" src="https://github.com/user-attachments/assets/914b1efb-f4df-4b45-8d17-f98023ccadd9">

### The upgrade fixes the alignment on the radio button conditional reveal
<img width="569" alt="image" src="https://github.com/user-attachments/assets/bbfcc7c7-a2a1-44a1-ba24-9db43a82bdfe">
<img width="569" alt="image" src="https://github.com/user-attachments/assets/475273bd-723b-4269-aac2-7a994bb0fd3e">

### I fixed a misalignment of the Breadcrumb chevron on mobile
Feel free to critique as I eyeballed the alignment
<img width="279" alt="image" src="https://github.com/user-attachments/assets/64adf14f-a2d5-4aea-a5eb-63561d0194c2">

<img width="279" alt="image" src="https://github.com/user-attachments/assets/585ecb68-14af-40a1-ad9c-3fc8d42ee0d3">

andysellick commented on package.json:
Question I think I already know the answer to - does this PR contain two separate commits for upgrading `govuk-frontend` because the commits that follow them relate to that version upgrade?

AshGDS replied:
@andysellick Yep - I initially looked at upgrading to exactly `5.4.0` as I was being strict to follow the trello card, but then realised it's probably more efficient to do the `5.5.0` upgrade in this PR as well.

andysellick commented on app/assets/stylesheets/govuk_publishing_components/components/helpers/_govuk-frontend-settings.scss:
```diff
@@ -13,3 +13,4 @@
 
 $govuk-image-url-function: "app-image-url";
 $govuk-font-url-function: "app-font-url";
+$govuk-new-typography-scale: true;
```
Nitpick: it might help clarify some of these changes if the commit message included a link to the change in `govuk-frontend` that they refer to e.g. in this case https://github.com/alphagov/govuk-frontend/blob/main/CHANGELOG.md#weve-adjusted-our-responsive-type-scale

andysellick commented on CHANGELOG.md:
```diff
@@ -11,6 +11,7 @@
 
 * Ensure that organisation logos always print ([PR #4162](https://github.com/alphagov/govuk_publishing_components/pull/4162))
 * Fix some layout_header search visual bugs ([PR #4156](https://github.com/alphagov/govuk_publishing_components/pull/4156))
+* Upgrade to govuk-frontend v5.5.0 ([PR #4115](https://github.com/alphagov/govuk_publishing_components/pull/4160))
```
Wondering if this is a breaking change? If it was deployed without the fix in `finder-frontend` it would cause something to look broken.

andysellick reviewed (commented):
Looks good, just a few minor comments.

The description mentions the change in https://github.com/alphagov/govuk_publishing_components/pull/3991/commits, is it worth cherry picking that change into this PR?

AshGDS commented:
Thanks Andy, I have:

1. Included Matt's commit
2. Added references to the `govuk-frontend` changelog in the commits
3. Switched it to a breaking change

Should be ready for another review now 👍

andysellick commented on app/views/govuk_publishing_components/components/_breadcrumbs.html.erb:
Nitpick: the URL in this commit message is wrong, I think it should be https://github.com/alphagov/govuk-frontend/blob/main/CHANGELOG.md#update-breadcrumbs-to-use-nav-and-aria-label

AshGDS replied:
@andysellick Thanks, fixed 👍
